pentasharp/Controllers/TaxiReservationController.cs

```diff
@@ -0,0 +1,188 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using pentasharp.Data;
+using pentasharp.Models.Entities;
+using pentasharp.Models.Enums;
+using pentasharp.ViewModel.TaxiModels;
+using pentasharp.ViewModel.TaxiReservation;
+using WebApplication1.Filters;
+
+namespace pentasharp.Controllers
+{
+    [Route("api/TaxiReservation")]
+    public class TaxiReservationController : Controller
+    {
+        private readonly AppDbContext _context;
+
+        public TaxiReservationController(AppDbContext context)
+        {
+            _context = context;
+        }
+
+        [HttpPost("SearchAvailableTaxis")]
+        public IActionResult SearchAvailableTaxis()
+        {
+            var taxiCompanies = _context.TaxiCompanies
+                .Select(c => new
+                {
+                    c.TaxiCompanyId,
+                    c.CompanyName,
+                    c.ContactInfo
+                })
+                .ToList();
+
+            return Ok(new { success = true, companies = taxiCompanies });
+        }
```

Comment:
add one white space in between APIs here and everywhere in your changes if it is missing